src/transformers/trainer.py

```diff
@@ -3376,20 +3376,17 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = Fa
                 version.parse(accelerate_version) > version.parse("0.24.1")
             ):
                 state_dict = self.accelerator.get_state_dict(self.model)
-                if self.args.should_save:
-                    self._save(output_dir, state_dict=state_dict)
+                self._save(output_dir, state_dict=state_dict)
```

Comment:
This still doesn't quite get what Amy was after, and I agree it's the solution we want to look at:

> Update should_save to be true for all ranks in FSDP, DEEPSPEED+prompt tuning case

While yes that's not *quite* how it works for TPU, that difference is important IMO and we should do it this way.

Reply:
currently, should_save is a property of args, and controlled by many factors.
see
![image](https://github.com/user-attachments/assets/c1093023-9d9e-44df-86e2-e0fbe07f7540)

if we update should_save to be true for all ranks. we need other parameters to replace original should_save to control if we needs to save tokenizer and arguments.
https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L3529-L3533